joblib/test/common.py

```diff
@@ -17,20 +18,46 @@
     import numpy as np
 
     def with_numpy(func):
-        """ A decorator to skip tests requiring numpy.
-        """
+        """A decorator to skip tests requiring numpy."""
         return func
 
 except ImportError:
     def with_numpy(func):
-        """ A decorator to skip tests requiring numpy.
-        """
+        """A decorator to skip tests requiring numpy."""
         def my_func():
             raise nose.SkipTest('Test requires numpy')
         return my_func
     np = None
 
 
+# we use memory_profiler library for memory consumption checks
+try:
+    from memory_profiler import memory_usage
+
+    def with_memory_usage(func):
+        """A decorator to skip tests requiring memory_profiler."""
+        return func
+
+    def memory_used(func, *args, **kwargs):
+        """Compute memory usage when executing func."""
+        gc.collect()
+        ref_mem = memory_usage(max_usage=True)
+        mem_use = memory_usage((func, args, kwargs),
+                               interval=.001, timeout=1, max_usage=True)
+        return mem_use[0] - ref_mem
+
+except ImportError:
+    def with_memory_usage(func):
```

Comment:
Done (partially) because the memory profile test functions are not fully working in nilearn. I'm not testing the low threshold with memory usage, as it's a real pain to make this work with memory_profiler (even with a memory offset).